### DISCUSSION TITLE
why select.add_columns() no work?

### DISCUSSION THREAD

**Discussion by mydu27** [sqlalchemy/sqlalchemy#8149] [Usage Questions] [answered]
I have a User model, sometimes I only need two columns(id, phonenumber) in User model, so I use add_columns() to resolve it, like this:
![image](https://user-images.githubusercontent.com/37979204/174544588-09fe8cf4-00be-4d7d-9be2-60f90368aca0.png)
![image](https://user-images.githubusercontent.com/37979204/174544629-c4285f1a-f2e1-4cf6-bfbf-85692869c2c9.png)
My expect result is got two columns. But I got an id list……What wrong？my sqlalchemy version is 1.4.36

**Comment by CaselIT** [comment, marked as answer]
Hi,
You are using `result.scalars()` to obtain only the fist column as a list

**Reply by mydu27** [reply]
> Hi, You are using `result.scalars()` to obtain only the fist column as a list
Yes, thanks. It's effective, but I still have a question, if I use `result.all()` to get the result , it will return a tuple list. How can I get a model object list like `result.scalars().all()`?

**Reply by CaselIT** [reply]
You are selecting columns if you do `select(self.model.id)` so the return value are python int, str etc in a tuple, representing then values of the selected columns.

If you want to get return orm object with only some column loaded, you should instead use the loader options [`load_only`](https://docs.sqlalchemy.org/en/14/orm/loading_columns.html?highlight=load_only#sqlalchemy.orm.load_only)
so your case would be something like `select(self.model).options(load_only(self.model.id, self.model.phoneNumber)`. You would then use `result.scalars()` to get a list of `self.model` instances

**Reply by mydu27** [reply]
> 如果这样做，则选择列`select(self.model.id)`，返回值是元组中的 python int、str 等，表示所选列的值。
> 
> 如果您想获得仅加载某些列的返回 orm 对象，则应改为使用加载器选项 ，以便您的情况类似于. 然后，您将使用获取实例列表[`load_only`](https://docs.sqlalchemy.org/en/14/orm/loading_columns.html?highlight=load_only#sqlalchemy.orm.load_only)`select(self.model).options(load_only(self.model.id, self.model.phoneNumber)``result.scalars()``self.model`

Really appreciate your help. And after reading the document again, I can also use `result.mapping()` to get the similar result. I learned a lot from your answer , thanks

**Reply by CaselIT** [reply]
Great, glad you resolved!
